### PR TITLE
Add Custom Network during on boarding

### PR DESCRIPTION
**Pull request by tommasini**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**
Now it's possible to add a custom network during onboarding, that enables the user to choose not to use a infura rpc url.

**Screenshots/Recordings**
https://recordit.co/PIQjzWnaue

**Test Cases**
Case1:
* Fresh install the app
* Press settings on meta metrics screen
* add another rpc url for ethereum
* finish on boarding (create a new wallet and importing an existing one) 
* go to settings -> networks
* check active network and active network rpc url

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by cortisiko** [comment]
this is not ready for QA since it did not go through dev review. Once we pass dev review. Feel free to add back the "needs-qa" label.

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js [hunk 1]
```diff
@@ -280,6 +307,11 @@ class NetworkSettings extends PureComponent {
         editable = true;
         rpcUrl = network;
       }
+      // Override values if UI is
+      if (isRPCUpdate) {
+        nickname = DEFAULT_MAINNET_NAME;
```
Comment (sethkfman) [inline app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js:312]: I think we should add a DEFAULT_MAINNET_CUSTOM_NAME = Ethereum Main Custom. Using the existing name could cause confusion

File: app/constants/network.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ export const GOERLI = 'goerli';
 export const RPC = 'rpc';
 export const NO_RPC_BLOCK_EXPLORER = 'NO_BLOCK_EXPLORER';
 export const PRIVATENETWORK = 'PRIVATENETWORK';
+export const DEFAULT_MAINNET_NAME = 'Default Main Network';
```
Comment (sethkfman) [inline app/constants/network.js:10]: Consider updating per comment where value is used.

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by cortisiko — CHANGES_REQUESTED** [review]
@Cal-L I am done with testing. Below are my findings:

**issue 1**
from the MetaMetrics view if i were to open the custom RPC form:
enter an RPC url
hit save/confirm/set. basically, tapping the CTA confirms the RPC url.
i am taken back to the MetaMetrics view.
if i were to return back to the custom RPC url form, the saved RPC URL does not appear: http://recordit.co/lEz0ONEIZa

**Issue 2**
Custom RPC for mainnet edit the RPC after wallet creation and confirm you are not seeing duplicate networks. See recording: http://recordit.co/owxMG7DvX6

**Issue 3**
This one is sort of sensitive. [Here is the slack chat](https://consensys.slack.com/archives/C04E6UR5WEM/p1670277127391459?thread_ts=1670271057.998489&cid=C04E6UR5WEM) with the bug description.

**Comment by Cal-L** [comment]
Closing and moved to #5362

**Pull request closed without merging**